File: backend/src/main/java/com/web/backend/model/atividade/Atividade.java
package com.web.backend.model.atividade;

import com.web.backend.model.Midia;
import jakarta.persistence.*;

import java.util.ArrayList;

@Entity
@Table(name = "atividade")
public class Atividade {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(columnDefinition = "text")
    private String nome;

    @Column(columnDefinition = "integer")
    private int duracaoMinutos;

    @ManyToMany(cascade = {CascadeType.PERSIST, CascadeType.MERGE})
    @JoinTable(
            name = "atividade_midia",
            joinColumns = @JoinColumn(name = "atividade_id"),
            inverseJoinColumns = @JoinColumn(name = "midia_id")
    )
    private ArrayList<Midia> imagens;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public int getDuracaoMinutos() {
        return duracaoMinutos;
    }

    public void setDuracaoMinutos(int duracaoMinutos) {
        this.duracaoMinutos = duracaoMinutos;
    }

    public void addImagem(Midia midia) {
        this.imagens.add(midia);
    }

    public ArrayList<Midia> getImagens() {
        return imagens;
    }

    public void setImagens(ArrayList<Midia> imagens) {
        this.imagens = imagens;
    }
}

File: frontend/src/model/Atividade.ts
class Atividade {
    private _id: number;
    private _nome: string;
    private _duracaoMinutos: number;

    constructor(id: number, nome: string, duracaoMinutos: number) {
        this._id = id;
        this._nome = nome;
        this._duracaoMinutos = duracaoMinutos;
    }

    get id(): number {
        return this._id;
    }

    set id(value: number) {
        this._id = value;
    }

    get nome(): string {
        return this._nome;
    }

    set nome(value: string) {
        this._nome = value;
    }

    get duracaoMinutos(): number {
        return this._duracaoMinutos;
    }

    set duracaoMinutos(value: number) {
        this._duracaoMinutos = value;
    }
}

export default Atividade